mypy/plugins/dataclasses.py

```diff
@@ -204,7 +204,10 @@ def collect_attributes(self) -> List[DataclassAttribute]:
                 continue
 
             node = cls.info.names[lhs.name].node
-            assert isinstance(node, Var)
+            assert isinstance(node, (Var, OverloadedFuncDef))
+
+            if isinstance(node, OverloadedFuncDef):
+                continue
```

Comment:
This is still dangerous, what if `node` is a `Decorator`? I would just replace the old assert with `if not isinstance(node, Var): continue`.